### PR TITLE
:sparkles: Add dynamic version badge if package exists on NPM

### PR DESCRIPTION
**Pull request by SwapnikKatkoori**
Referencing Issue #54 

If a package exists on NPM, it adds the dynamic NPM version badge. If it does not exist on NPM, the version badge stays as the default static version.

Changes:
- Added a new function in project-infos.js called getVersionBadgeInfo() that returns an object of the for {type: string, data: string }.
     - The type will either be "dynamic", if the package exists on NPM, and "static" if it does not. If the type is "static" the data will be the version number like before. If the type is "dynamic", the data will be the name of the package.

- In the markdown templates, the badge is set based on the version badge info.

- Updated the tests the reflect the new changes to "version" in  project-infos.

- Changed the snapshot. 

Testing:

- Used npm link and a test repo to test the badges.
     - Used a package name that existed on NPM to ensure that a dynamic badge was generated.
     - Used a  package name that did not exist on NPM to ensure that a dynamic badge was generated.

Please let me know if any of this needs to be changed.

### CODE REVIEW

**Comment by SwapnikKatkoori** [comment]
It looks like I did not do the snapshot testing correctly. Any ideas @kefranabg (or anyone)?

File: package-lock.json [hunk 1]
```diff
@@ -3055,8 +3055,7 @@
         "ansi-regex": {
           "version": "2.1.1",
           "bundled": true,
-          "dev": true,
```
Comment (kefranabg) [inline package-lock.json]: `package-lock.json` should not be modified by a PR 😉

File: src/project-infos.js [hunk 2]
```diff
@@ -113,6 +113,29 @@ const getAuthorName = packageJson => {
...
+/**
+ * Returns an object with dynamic badge information if the package exists on NPM.
+ * It returns data for static badge if it does not exist on NPM.
+ *
+ * @param projectName
+ * @param packageJson
+ * @returns {Object{type:string, data:string}}
+ * Type of badge can be "static" or "dynamic". Data can be undefined.
+ */
+const getVersionBadgeInfo = (projectName, packageJson) => {
```
Comment (kefranabg) [inline src/project-infos.js]: I think that returning an object is a bit overkill. The only information we need to get is if project is available on npm or not. We should have a function named `isProjectAvailableOnNpm` that return a boolean. This boolean would be used in the ejs template to tell if we render a static or dynamic badge

File: src/project-infos.js [hunk 3]
```diff
@@ -124,7 +147,7 @@ const getProjectInfos = async () => {
   const description = get(packageJson, 'description', undefined)
   const engines = get(packageJson, 'engines', undefined)
   const author = getAuthorName(packageJson)
-  const version = get(packageJson, 'version', undefined)
```
Comment (kefranabg) [inline src/project-infos.js]: This should not be changed, you can add a new information instead on modifying existing one 😉

File: templates/default-no-html.md [hunk 4]
```diff
@@ -1,6 +1,9 @@
 # Welcome to <%= projectName %> 👋
-<% if (projectVersion) { -%>
-![Version](https://img.shields.io/badge/version-<%= projectVersion %>-blue.svg?cacheSeconds=2592000)
+<% if (projectVersion.type === "static" && projectVersion.data) { -%>
```
Comment (kefranabg) [inline templates/default-no-html.md]: As mentioned before, the rendering should depends on a boolean property instead of a type

**Comment by SwapnikKatkoori** [comment]
@kefranabg, In commit df3c230 :
- I reverted the package-lock.json back to the one from origin/master
- Changed the function to isProjectAvailableOnNpm() and it returns a boolean. true if it is avaibale on NPM, and false if it is not. 
- I changed the version variable to what it was before the initial commit, and added a new bool variable called isProjectOnNpm.
- Updated the unit tests to account for isProjectOnNpm.

I hope I got everything as per your suggestions above. Let me know if there is anything else you want me to change. 

I am also still having trouble getting the check-prettier and snapshot tests to pass. So if you have any info on that, please let me know.

**Pull request merged**